lstchain/scripts/lstchain_dl1ab.py

```diff
@@ -211,29 +212,24 @@ def main():
             parameters_to_update.extend(uncertainty_params)
 
         if increase_nsb:
-            rng = np.random.default_rng(
-                    infile.root.dl1.event.subarray.trigger.col('obs_id')[0])
+            rng = np.random.default_rng(infile.root.dl1.event.subarray.trigger.col('obs_id')[0])
 
         if increase_psf:
             set_numba_seed(infile.root.dl1.event.subarray.trigger.col('obs_id')[0])
 
         image_mask_save = not args.no_image and 'image_mask' in infile.root[dl1_images_lstcam_key].colnames
+        if image_mask_save:
+            image_mask = image_table['image_mask']
+
+        params = read_table(infile, dl1_params_lstcam_key)
+        new_params = set(parameters_to_update) - set(params.colnames)
+        for p in new_params:
+            params[p] = np.empty(len(params), dtype=np.float32)
```

Comment:
Looking at the list of `parameters_to_update`, not all of them are float32, there is for example `n_pixels`.

Reply:
Well, how about something like this:
```
parameter_dtypes = {
    "n_pixels":  np.int32,
    "n_islands":  np.int32,
}


...


params[p] = np.empty(len(params), dtype=parameter_dtypes.get(p, np.float32)`
```

Reply:
that's a possibility but not so flexible if new parameters are added.
what about guessing types from the dl1_container?

Reply:
You could look at `type(d1_container.<field>.default)`, yes.

Reply:
However, it seems these are using python values / float64 at the moment, which should probably be corrected...

Reply:
Thanks @maxnoe .
Following this, I made the corresponding changes, setting all types in `DL1ParametersContainer`

Reply:
Was `float64` really necessary for the `intensity` and `log_intensity` ?